packages/rest-typings/src/v1/groups/GroupsMembersByOrderedRole.ts

```diff
@@ -0,0 +1,59 @@
+import type { IRole, IRoom } from '@rocket.chat/core-typings';
+
+import type { PaginatedRequest } from '../../helpers/PaginatedRequest';
+import { ajv } from '../Ajv';
+
+type MembersOrderedByRoleProps = {
+	roomId?: IRoom['_id'];
+	roomName?: IRoom['name'];
+	status?: string[];
+	filter?: string;
+	rolesOrder?: IRole['_id'][];
+};
+
+export type GroupsMembersOrderedByRoleProps = PaginatedRequest<MembersOrderedByRoleProps>;
+
+const membersOrderedByRoleRolePropsSchema = {
+	properties: {
+		roomId: {
+			type: 'string',
+		},
+		roomName: {
+			type: 'string',
+		},
+		rolesOrder: {
+			type: 'array',
+			items: {
+				type: 'string',
```

Comment:
I think we need to support an enum here OR at least validate the values where we process it and not to allow the users to send arbitrary values here.

Reply:
The roles can be custom roles added by the admin. We can check the values if they exists in the db. wdys ?